python/ray/tests/test_failure_2.py

```diff
@@ -531,7 +531,8 @@ def test_raylet_node_manager_server_failure(ray_start_cluster_head,
     cluster = ray_start_cluster_head
     redis_port = int(cluster.address.split(":")[1])
     # Reuse redis port to make node manager grpc server fail to start.
-    cluster.add_node(wait=False, node_manager_port=redis_port)
+    with pytest.raises(TimeoutError):
```

Comment:
I am a little confused about this change. How is the behavior completely changed from before?

Reply:
Please see comments here: https://github.com/ray-project/ray/pull/16045/files#diff-91cb97f6f4bd37eeebc5508219672f5bc676c75275e97dd603ae8bc3e4defb3aR244